### ISSUE TITLE
Steps to get rid of the Twig experimental state

### ISSUE THREAD

**Issue by m-vo** [contao/contao#3566]
In this issue, I'd like to track things that should happen with our Twig implementation become it can become part of the BC promise. I'll update the issue over time.

* [x] #3274 
* [x] Consider making the `ContextHelper` a service
* [ ] Remove `@experimental`, review what needs to become `@internal`
* [x] Refresh the template cache when editing Twig templates in the back end
* [x] Allow creating Twig templates from the back end.
* [ ] …

**Comment by aschempp** [comment]
should we also have support to edit Twig templates in the back end? Or is that already given?

**Comment by m-vo** [comment]
> should we also have support to edit Twig templates in the back end? Or is that already given?

This should already work, we only need to add some logic that refreshes the template cache once saved. I added this to the list.

**Comment by aschempp** [comment]
I can edit files that live in `/templates`, but I cannot choose Twig templates to add a new template override in my root folder.

**Comment by m-vo** [comment]
Correct, that needs to be added. But we'll have to rework this whole template selector anyways so that you will get the right templates.

**Comment by aschempp** [comment]
I know. But it should be on this list, right? 😉

**Comment by fkaminski** [comment]
Just a reminder.
If you put {{ dump() }} in the Twig template it works in debug mode. Otherwise you get an error message:

`Uncaught PHP Exception Twig\Error\SyntaxError: "Did you forget to run "composer require symfony/debug-bundle"?"`

**Comment by m-vo** [comment]
> If you put {{ dump() }} in the Twig template it works in debug mode. Otherwise you get an error message:

Cannot confirm this. Please make sure the `symfony/twig-bridge` got installed in your application and open up an issue if the error still persists after a `composer update`. The `dump()` function should return nothing in production but not fail.

**Comment by fritzmg** [comment]
I can confirm the problem. Using

```twig
{% extends '@Contao/fe_page' %}

{% block main %}
  {{ dump('foo') }}
  <h1>Hello from Twig!</h1>
  {{ parent() }}
{% endblock %}
```

will yield

```
[2021-12-21T10:54:23.936147+00:00] request.CRITICAL: Uncaught PHP Exception Twig\Error\SyntaxError: "Did you forget to run "composer require symfony/debug-bundle"? Unknown function "dump" in "@Contao/fe_page.html.twig"." at vendor\symfony\twig-bridge\UndefinedCallableHandler.php line 96 {"exception":"[object] (Twig\\Error\\SyntaxError(code: 0): Did you forget to run \"composer require symfony/debug-bundle\"? Unknown function \"dump\" in \"@Contao/fe_page.html.twig\". at vendor\\symfony\\twig-bridge\\UndefinedCallableHandler.php:96)"} []
```

This error is in fact thrown by `symfony/twig-bridge`.
